src/briefcase/platforms/android/gradle.py

```diff
@@ -182,7 +190,7 @@ def build_app(self, app: BaseConfig, **kwargs):
         try:
             env = {**self.os.environ, "ANDROID_SDK_ROOT": str(self.sdk_path)}
             self.subprocess.run(
-                ["./gradlew", "assembleDebug"],
+                [str(self.gradlew_path(app)), "assembleDebug"],
                 env=env,
                 cwd=str(self.bundle_path(app)),
```

Comment:
In my testing, it is needed; otherwise, gradle doesn't find the app configuration. I didn't record _excellent_ evidence of this; this is from memory; but that is what I remember.

The best thing for me to do would be to confirm this, then add a comment indicating it's true. I'll do that shortly.

Reply:
Done & pushed to this PR! Marking resolved.

Reply:
Ah - right - this is "run gradle in your project directory". Having a prefix on the gradle invocation is what threw me off the scent; I know the explicit path is no less *correct* - but I wonder if it's mildly misleading.

Reply:
What do you think of the new comment?

https://github.com/beeware/briefcase/pull/336/files#diff-a8d342005831b1bde23705e260e93a38R195

I could also add a comment explaining why we're specifying the full path to `gradlew`.

On macOS & Linux, I can use `./gradlew`. On Windows, however, both of the following (when passed to `subprocess`) raise `FileNotFoundError`:

- `./gradlew.bat`
- `.\\gradlew.bat`

even with the `cwd=`.

For this reason, I think it's simplest to specify the full path. Do you think I should add a further comment explaining why we pass the full path to `gradlew`? Happy to, if so.

Once we figure this out, we can merge this 🎉 !!! I'm excited!

Reply:
Is the ./ prefix required at all on windows? local directory should be an implied path, IIRC

Reply:
(and yes - the new comment is really good!)

Reply:
`gradlew.bat` also raises `FileNotFoundError`. I think this is due to `subprocess` having a different, more explicit `%PATH%`-based resolution compared to `cmd.exe`.